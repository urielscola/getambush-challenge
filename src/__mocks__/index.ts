export * from './requests';
export * from './utils';
