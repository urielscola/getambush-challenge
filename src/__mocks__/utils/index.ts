export * from './withExternalDependencies';
export * from './mockApi';
