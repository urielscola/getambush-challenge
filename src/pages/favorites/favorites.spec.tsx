import React from 'react';
import { render, act, waitFor, RenderResult } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { withExternalDependencies } from '__mocks__';
import Favorites from './index';

describe('[PAGES] Favorites', () => {
  it('should render', async () => {
    let result: RenderResult;

    await act(async () => {
      result = render(withExternalDependencies(<Favorites />));
    });

    await waitFor(() => {
      expect(result).toMatchSnapshot();
    });
  });
});
