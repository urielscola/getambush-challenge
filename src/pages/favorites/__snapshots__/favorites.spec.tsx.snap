// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PAGES] Favorites should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-jrQzAO iQNJRr"
      >
        <nav
          class="sc-iqseJM gA-Dvch"
        >
          <div
            class="sc-bdvvtL ipwLvv"
          >
            <div
              class="sc-gsDKAQ jYZISc"
              data-testid="flex-div"
              display="flex"
            >
              <ul>
                <li>
                  <a
                    appearence="primary"
                    aria-disabled="false"
                    class="sc-pVTFL yywlw"
                    data-testid="button"
                    href="/favorites"
                    rounded="false"
                    tabindex="0"
                  >
                    Favorites
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </nav>
        <div
          class="sc-kDTinF jARzRY"
        >
          <div
            class="sc-gsDKAQ ibRzVS"
            data-testid="flex-div"
            display="flex"
          >
            <a
              href="/"
            >
              <div
                class="sc-gsDKAQ eZPjUy"
                data-testid="flex-div"
                display="flex"
              >
                <h1
                  class="sc-eCImPb bkIxqH"
                  color="#fff"
                  data-testid="text"
                >
                  GitHub Search
                </h1>
                <div
                  aria-hidden="true"
                  class="sc-dkPtRN eVSKHq"
                >
                  <svg
                    aria-hidden="true"
                    class="StyledIconBase-ea9ulj-0 bsimgx"
                    color="#788cfe"
                    fill="currentColor"
                    focusable="false"
                    height="40"
                    viewBox="0 0 24 24"
                    width="40"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12.026 2c-5.509 0-9.974 4.465-9.974 9.974 0 4.406 2.857 8.145 6.821 9.465.499.09.679-.217.679-.481 0-.237-.008-.865-.011-1.696-2.775.602-3.361-1.338-3.361-1.338-.452-1.152-1.107-1.459-1.107-1.459-.905-.619.069-.605.069-.605 1.002.07 1.527 1.028 1.527 1.028.89 1.524 2.336 1.084 2.902.829.091-.645.351-1.085.635-1.334-2.214-.251-4.542-1.107-4.542-4.93 0-1.087.389-1.979 1.024-2.675-.101-.253-.446-1.268.099-2.64 0 0 .837-.269 2.742 1.021a9.582 9.582 0 0 1 2.496-.336 9.554 9.554 0 0 1 2.496.336c1.906-1.291 2.742-1.021 2.742-1.021.545 1.372.203 2.387.099 2.64.64.696 1.024 1.587 1.024 2.675 0 3.833-2.33 4.675-4.552 4.922.355.308.675.916.675 1.846 0 1.334-.012 2.41-.012 2.737 0 .267.178.577.687.479C19.146 20.115 22 16.379 22 11.974 22 6.465 17.535 2 12.026 2z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </div>
            </a>
            <div
              class="sc-gsDKAQ eZPjUy"
              data-testid="flex-div"
              display="flex"
            >
              <p
                class="sc-eCImPb jaXENj"
                data-testid="text"
              >
                Your Favorites
              </p>
              <div
                aria-hidden="true"
                class="sc-dkPtRN eVSKHq"
              >
                <svg
                  aria-hidden="true"
                  class="StyledIconBase-ea9ulj-0 bsimgx"
                  color="#e35b5c"
                  fill="currentColor"
                  focusable="false"
                  height="40"
                  viewBox="0 0 16 16"
                  width="40"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 1.314C12.438-3.248 23.534 4.735 8 15-7.534 4.736 3.562-3.248 8 1.314z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </header>
      <main>
        <div
          class="sc-bdvvtL ipwLvv"
        >
          <div
            class="sc-gsDKAQ fGHhBO"
            data-testid="flex-div"
            display="flex"
          >
            <div
              class="sc-gsDKAQ gCZdIE"
              data-testid="flex-div"
              display="flex"
            >
              <p
                class="sc-eCImPb hFcoVp"
                data-testid="text"
              >
                No favorites.
              </p>
            </div>
          </div>
        </div>
      </main>
      <footer
        class="sc-crHmcD cfnhne"
      >
        <div
          class="sc-bdvvtL ipwLvv"
        >
          <p
            class="sc-eCImPb mqDDb"
            data-testid="text"
          >
            CC 2021
          </p>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-jrQzAO iQNJRr"
    >
      <nav
        class="sc-iqseJM gA-Dvch"
      >
        <div
          class="sc-bdvvtL ipwLvv"
        >
          <div
            class="sc-gsDKAQ jYZISc"
            data-testid="flex-div"
            display="flex"
          >
            <ul>
              <li>
                <a
                  appearence="primary"
                  aria-disabled="false"
                  class="sc-pVTFL yywlw"
                  data-testid="button"
                  href="/favorites"
                  rounded="false"
                  tabindex="0"
                >
                  Favorites
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
      <div
        class="sc-kDTinF jARzRY"
      >
        <div
          class="sc-gsDKAQ ibRzVS"
          data-testid="flex-div"
          display="flex"
        >
          <a
            href="/"
          >
            <div
              class="sc-gsDKAQ eZPjUy"
              data-testid="flex-div"
              display="flex"
            >
              <h1
                class="sc-eCImPb bkIxqH"
                color="#fff"
                data-testid="text"
              >
                GitHub Search
              </h1>
              <div
                aria-hidden="true"
                class="sc-dkPtRN eVSKHq"
              >
                <svg
                  aria-hidden="true"
                  class="StyledIconBase-ea9ulj-0 bsimgx"
                  color="#788cfe"
                  fill="currentColor"
                  focusable="false"
                  height="40"
                  viewBox="0 0 24 24"
                  width="40"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12.026 2c-5.509 0-9.974 4.465-9.974 9.974 0 4.406 2.857 8.145 6.821 9.465.499.09.679-.217.679-.481 0-.237-.008-.865-.011-1.696-2.775.602-3.361-1.338-3.361-1.338-.452-1.152-1.107-1.459-1.107-1.459-.905-.619.069-.605.069-.605 1.002.07 1.527 1.028 1.527 1.028.89 1.524 2.336 1.084 2.902.829.091-.645.351-1.085.635-1.334-2.214-.251-4.542-1.107-4.542-4.93 0-1.087.389-1.979 1.024-2.675-.101-.253-.446-1.268.099-2.64 0 0 .837-.269 2.742 1.021a9.582 9.582 0 0 1 2.496-.336 9.554 9.554 0 0 1 2.496.336c1.906-1.291 2.742-1.021 2.742-1.021.545 1.372.203 2.387.099 2.64.64.696 1.024 1.587 1.024 2.675 0 3.833-2.33 4.675-4.552 4.922.355.308.675.916.675 1.846 0 1.334-.012 2.41-.012 2.737 0 .267.178.577.687.479C19.146 20.115 22 16.379 22 11.974 22 6.465 17.535 2 12.026 2z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </a>
          <div
            class="sc-gsDKAQ eZPjUy"
            data-testid="flex-div"
            display="flex"
          >
            <p
              class="sc-eCImPb jaXENj"
              data-testid="text"
            >
              Your Favorites
            </p>
            <div
              aria-hidden="true"
              class="sc-dkPtRN eVSKHq"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 bsimgx"
                color="#e35b5c"
                fill="currentColor"
                focusable="false"
                height="40"
                viewBox="0 0 16 16"
                width="40"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 1.314C12.438-3.248 23.534 4.735 8 15-7.534 4.736 3.562-3.248 8 1.314z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </header>
    <main>
      <div
        class="sc-bdvvtL ipwLvv"
      >
        <div
          class="sc-gsDKAQ fGHhBO"
          data-testid="flex-div"
          display="flex"
        >
          <div
            class="sc-gsDKAQ gCZdIE"
            data-testid="flex-div"
            display="flex"
          >
            <p
              class="sc-eCImPb hFcoVp"
              data-testid="text"
            >
              No favorites.
            </p>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="sc-crHmcD cfnhne"
    >
      <div
        class="sc-bdvvtL ipwLvv"
      >
        <p
          class="sc-eCImPb mqDDb"
          data-testid="text"
        >
          CC 2021
        </p>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
